perception/euclidean_cluster/src/euclidean_cluster_node.cpp

```diff
@@ -48,23 +47,10 @@ void EuclideanClusterNode::onPointCloud(
   pcl::fromROSMsg(*input_msg, *raw_pointcloud_ptr);
 
   // clustering
-  std::vector<pcl::PointCloud<pcl::PointXYZ>> clusters;
-  cluster_->cluster(raw_pointcloud_ptr, clusters);
-
-  // build output msg
-  tier4_perception_msgs::msg::DetectedObjectsWithFeature output;
-  convertPointCloudClusters2Msg(input_msg->header, clusters, output);
-  cluster_pub_->publish(output);
-
-  // build debug msg
-  if (debug_pub_->get_subscription_count() < 1) {
-    return;
-  }
-  {
-    sensor_msgs::msg::PointCloud2 debug;
-    convertObjectMsg2SensorMsg(output, debug);
-    debug_pub_->publish(debug);
-  }
+  std::shared_ptr<autoware_auto_perception_msgs::msg::DetectedObjects> detected_objects_ptr = cluster_->cluster(raw_pointcloud_ptr);
+  detected_objects_ptr->header.frame_id = input_msg->header.frame_id;
+  detected_objects_ptr->header.stamp = input_msg->header.stamp;
+  detected_objects_pub_->publish(*detected_objects_ptr);
```

Comment:
Again, unless you have specific reasons for using the ROS message inside the `cluster_` object, please change this back and have the messages populated here.

Reply:
addressed in https://github.com/upowerrobotics/autoware.universe/pull/4/commits/4c625fbeaad3c87b187ebb30d3caf125d75be5e5